src/ClusterBootstrap/cloud_init_aztools.py

```diff
@@ -334,6 +334,9 @@ def add_machine_in_parallel(cmds, args):
 
 
 def add_machine(vmname, spec, verbose, dryrun, output_file):
+    multual_exclusive_roles = set(["infra", "worker", "elasticsearch", "mysqlserver"])
```

Comment:
We need to refactor his function in a separate PR. The logic has become somewhat untrackable.

Reply:
sure, plz let me know the preferred way